---
title: "Jumps Analysis in Target Sequences"
author: "Qian Feng"
date: "2018/7/13"
output:
  pdf_document: default
---

```{r global_options, include=FALSE}
knitr::opts_chunk$set(fig.width=12, fig.height=8,
                      echo=TRUE, warning=FALSE, message=FALSE,
                      tidy=TRUE)
```

## Load R libraries
```{r, echo=FALSE, message=FALSE, warning=FALSE}
library(data.table)
library(dplyr)
library(ggfortify)
library(ggplot2)
library(Rtsne)
library(stringr)
library(starmie)
library(ggtree)
library(ape)
library(pheatmap)
library(proxy)
library(stringr)
library(knitr)
library(plotly)
library(vioplot)
library(plyr)
library(Hmisc)
cols <- c("#a6cee3","#1f78b4","#b2df8a","#33a02c","#fb9a99","#e31a1c","#fdbf6f","#ff7f00","#cab2d6","#6a3d9a","#ffff99","#b15928","#8dd3c7","#ffffb3","#bebada","#fb8072","#80b1d3","#fdb462","#b3de69","#fccde5","#d9d9d9","#bc80bd","#ccebc5","#ffed6f")
```

## Investigate the recombination hotspots using the relative position

Let's try the global data from Gerry's paper firstly. 31946 sequences are involved.

The basic sequence length information is here:

Statistics      Value      
--------   --------------   
max        154               
min        12             
mean       124.72     
mode       125
median     125
variance   89.95
---------  --------------   


```{python, eval=FALSE, include=FALSE}
import glob
from mungo.fasta import FastaReader
from collections import defaultdict
import numpy as np
from __future__ import division
mapping_dict = {}
with open("/Users/fengqian/global_var/qian_report/mosaic_processed_data/Protein_NoLab_translateable_combined_454_tessema_centroids_reducedRegion.fasta_mapping.txt", 'rU') as infile:
    for line in infile:
        line=line.strip().split()
        mapping_dict[line[0]]=line[1]

target_jumps = defaultdict(list)
target_jumps_proportions = []
for align_file in glob.glob("/Users/fengqian/global_var/qian_report/mosaic_processed_data/results_full/*.fasta_align.txt"):
    with open(align_file, 'rU') as infile:
        for line in infile:
            if "target_seq" in line:
                if "Target:" in line: continue
                pos = 0
                gaps = 0
                line = line.strip().split()
                read = mapping_dict[line[0].split("_")[1]]
                seq = line[1]
                seq_actual=len(seq)-seq.count("-")
            if "db_seq" in line:
                align_len = len(line.strip().split()[1])
                pos += align_len
                if pos>=len(seq): continue #we've got to the end
                gaps += seq[(pos-align_len):pos].count("-")
                target_jumps[read].append(pos-gaps)
                target_jumps_proportions.append((pos-gaps)/seq_actual)
with open("/Users/fengqian/global_var/qian_report/mosaic_data/target_jumps_proportions.txt", 'w') as outfile:
    outfile.write(",".join([str(c) for c in target_jumps_proportions]))
```

```{r, echo=FALSE, message=FALSE, warning=FALSE}
recombination <- unlist(fread("/Users/fengqian/global_var/qian_report/mosaic_data/target_jumps_proportions.txt", data.table = FALSE)[1,])
recombination <- as.data.frame(table(recombination))
summary(recombination[,2]) ###frequencies summary
recombination[which(recombination[,2]==max(recombination[,2])),]#### location is right middle of target sequence.
plot(recombination,xlab="normalized position",ylab="jump_counts")
vioplot(recombination$Freq)
```

Let's take a close look at the distribution of jump frequencies, and on the contrary, look at the conserved jump counts.

```{r, echo=FALSE, message=FALSE, warning=FALSE}
plot(recombination,ylim=c(0,200),xlab="normalized position exclude 0.5",ylab="jump_counts")
lines(smooth.spline(recombination$recombination, recombination$Freq),col="blue")
hist(recombination$Freq,breaks=seq(0,628,1),axes=T,xlim=c(0,200),main ="Historgram of Freq exclude position 0.5",xlab="frequency",ylab="position counts")
```

This indicates that two ends of DBLa tags are the most conserved regions. At the middle region of tags jump happens more frequently. We have 31946 sequences, when the jump counts are compared to this value, it's still pretty small. Most positions in all sequences only jump once, most positions have less than 50 jumps,its maximum number should be the number of all sequences,namely 31946.

This is coded by Yao-ban, a cumulative jump frequency plot is obtained.
```{r, message=TRUE, warning=TRUE}
rec2 <- recombination
rec2$sumFreq <- rec2$Freq
for (i in 2:4787) { rec2$sumFreq[i] <- rec2$sumFreq[i] + rec2$sumFreq[i-1] }
plot(rec2$recombination, rec2$sumFreq, type="l")
```

Then let's look at its barplot.

```{r, echo=FALSE, message=FALSE, warning=FALSE}
position <- as.numeric(levels(recombination$recombination)[recombination$recombination])
recombination$cut <- cut2(position,seq(0,1,0.1))
rec_cumsum <- ddply(recombination, "cut",
                   transform, cumsums=cumsum(Freq))
temp <- as.data.frame(table(recombination$cut))
temp$sumFreq <- temp$Freq
for (i in 2:10) { temp$sumFreq[i] <- temp$sumFreq[i] + temp$sumFreq[i-1] }
```


```{r}
barplot(rec_cumsum[temp$sumFreq,4],names.arg=temp[,1],density=20,border="red",col="blue",xlab="Normalized position",ylab="Number of jumps",space=0)
abline(v=c(4.9,6.3),col="yellow")
abline(v=c(2.5,4.8),col="darkgreen")
```


```{r, echo=FALSE, message=FALSE, warning=FALSE}
ggplot(data=rec_cumsum, aes(x=recombination, y=cumsums)) +
  geom_bar(stat="identity")+
  geom_text(aes(y=cumsums, label=Freq), vjust=1.6, 
            color="yellow", size=3.5)+
  scale_fill_brewer(palette="Paired")+
  theme_minimal()
```


From the first bar plot, the position between [0.5,0.6) has the highest jump counts,followed by position range from 0.4 to 0.5. There is one decrease about the jump count in position at [0.3,0.4). Still, at the two endpoints of target sequences, they are silent regions which seledom occur jumps. Moreover, based on second plot, each block is cumulated, and the increasing trend is very interesting in each interval.[0.1,0.2) and [0.4,0.5) increase much more dramatically than other intervals. Two ends of targets sequences increase slowly.

Dr **Qixin He** told me more conserved region has a higher probability to occur recombination and share higher similarity sequence block. HB5 should be in range at position from [0.4,0.6),its related amino acid sequence starts from ``redww''. The position [0.2,0.3) is related with HB14.

Then let's look at the aligned protein sequences and unaligned sequence by AliView (seaview is also Ok). It seems ``redww'' is in the middle of sequences, HB2 is at the rightmost of sequences.The leftmost sequences are pretty conserved, but am not sure it's HB3. so I should look at the HB14. Moreover, I calculate the location of intervals for HB5 and HB14 in sequences.

The references for extracting HB5 and HB14 are below:

***

1. Rask, Thomas S., et al. "Plasmodium falciparum erythrocyte membrane protein 1 diversity in seven genomes, divide and conquer." PLoS computational biology 6.9 (2010): e1000933.
2. Larremore, Daniel B., et al. "Ape parasite origins of human malaria virulence genes." Nature communications 6 (2015): 8368.

***

```{python, eval=FALSE, include=FALSE}
##########extract HB5 positions in target sequence
import glob
import re
from mungo.fasta import FastaReader
from collections import defaultdict
import numpy as np
from __future__ import division
for h,s in FastaReader("/Users/fengqian/global_var/qian_report/mosaic_processed_data/Protein_NoLab_translateable_combined_454_tessema_centroids_reducedRegion.fasta"):
    L1=re.findall(r'(REDWW.*?\TC)',s);L2=re.findall(r'(REYWW.*?\TC)',s)
    if len("".join(L1))==0 and len("".join(L1))==0: continue
    elif len("".join(L1))!=0 and len("".join(L2))!=0:
        print ("extra HB5")
    else:
        L = L1+L2
        hb5_start = s.find("".join(L))+1
        hb5_end = hb5_start + len("".join(L))-1
        hb5_location = []
        hb5_location.append(hb5_start/len(s))
        hb5_location.append(hb5_end/len(s))
        with open("/Users/fengqian/global_var/qian_report/mosaic_data/hb5_location_analysis.txt", 'a') as outfile:
            outfile.write("\t".join([str(c) for c in hb5_location])+"\n")
```

```{r, echo=FALSE, message=FALSE, warning=FALSE}
HB5 <- fread("/Users/fengqian/global_var/qian_report/mosaic_data/hb5_location_analysis.txt"
                  , data.table = FALSE
                  , header=FALSE)
colnames(HB5)=c("left","right")
print("HB5 left endpoint location:")
summary(HB5$left)
print("HB5 right endpoint location:")
summary(HB5$right)
```
Therefore, HB5 is in the location between 49.1% to 62.7%, combined with previous barplot, this region has the highest number of jump counts.

```{python, eval=FALSE, include=FALSE}
##########extract HB14 positions in target sequence
import glob
import re
from mungo.fasta import FastaReader
from collections import defaultdict
import numpy as np
from __future__ import division
for h,s in FastaReader("/Users/fengqian/global_var/qian_report/mosaic_processed_data/Protein_NoLab_translateable_combined_454_tessema_centroids_reducedRegion.fasta"):
    L1=re.findall(r'(LK.*?\LREDWW)',s);L2=re.findall(r'(LQ.*?\LREDWW)',s)
    if len("".join(L1))==0 and len("".join(L1))==0: continue
    elif len("".join(L1))!=0 and len("".join(L2))!=0:###cover as many sites about HB14
        L = L2
        hb14_start = s.find("".join(L))+1
        hb14_end = hb14_start + len("".join(L))-1-5
        hb14_location = []
        hb14_location.append(hb14_start/len(s))
        hb14_location.append(hb14_end/len(s))
        with open("/Users/fengqian/global_var/qian_report/mosaic_data/hb14_location_analysis.txt", 'a') as outfile:
            outfile.write("\t".join([str(c) for c in hb14_location])+"\n")
    else:
        L = L1+L2
        hb14_start = s.find("".join(L))+1
        hb14_end = hb14_start + len("".join(L))-1-5
        hb14_location = []
        hb14_location.append(hb14_start/len(s))
        hb14_location.append(hb14_end/len(s))
        with open("/Users/fengqian/global_var/qian_report/mosaic_data/hb14_location_analysis.txt", 'a') as outfile:
            outfile.write("\t".join([str(c) for c in hb14_location])+"\n")
```

```{r, echo=FALSE, message=FALSE, warning=FALSE}
HB14 <- fread("/Users/fengqian/global_var/qian_report/mosaic_data/hb14_location_analysis.txt"
                  , data.table = FALSE
                  , header=FALSE)
colnames(HB14)=c("left","right")
print("HB14 left endpoint location:")
summary(HB14$left)
print("HB14 right endpoint location:")
summary(HB14$right)
```


In general, from 24.5% to 48.5% of sequence length, HB14 exists there, they are also in the region of high jump probabilities. Then region from 48.5% to 49.1% should be the highly variable region.






*********************************************************************************
*********************************************************************************
*********************************************************************************
# Ghana

Above is the analysis from global data, here we analyze te Ghana data specifily. There are 17335 sequences in total. The basic sequence length information is here:

Statistics      Value      
--------   --------------   
max        162               
min        42             
mean       124.85     
mode       122
median     124
variance   71.40
---------  --------------  

## Investigate the recombination hotspots using the relative position

```{python, eval=FALSE, include=FALSE}
import glob
from mungo.fasta import FastaReader
from collections import defaultdict
import numpy as np
from __future__ import division
mapping_dict = {}
with open("/Users/fengqian/Downloads/UniMelb_shared-master/project/mosaic_processed_data/Protein_translateable_pilot_upper_centroids.fasta_mapping.txt", 'rU') as infile:
    for line in infile:
        line=line.strip().split()
        mapping_dict[line[0]]=line[1]

target_jumps = defaultdict(list)
target_jumps_proportions = []
for align_file in glob.glob("/Users/fengqian/Downloads/UniMelb_shared-master/project/mosaic_processed_data/results_final_alignment/*.fasta_align.txt"):
    with open(align_file, 'rU') as infile:
        for line in infile:
            if "target_seq" in line:
                if "Target:" in line: continue
                pos = 0
                gaps = 0
                line = line.strip().split()
                read = mapping_dict[line[0].split("_")[1]]
                seq = line[1]
                seq_actual=len(seq)-seq.count("-")
            if "db_seq" in line:
                align_len = len(line.strip().split()[1])
                pos += align_len
                if pos>=len(seq): continue #we've got to the end
                gaps += seq[(pos-align_len):pos].count("-")
                target_jumps[read].append(pos-gaps)
                target_jumps_proportions.append((pos-gaps)/seq_actual)
with open("/Users/fengqian/Downloads/UniMelb_shared-master/project/mosaic_data/target_jumps_proportions.txt", 'w') as outfile:
    outfile.write(",".join([str(c) for c in target_jumps_proportions]))
```


## Now collect and analyze the breakpoint positions in each sequence.

```{r, echo=FALSE, message=FALSE, warning=FALSE}
recombination <- unlist(fread("/Users/fengqian/Downloads/UniMelb_shared-master/project/mosaic_data/target_jumps_proportions.txt", data.table = FALSE)[1,])
recombination <- as.data.frame(table(recombination))
summary(recombination[,2]) ###frequencies summary
recombination[which(recombination[,2]==max(recombination[,2])),]#### location is right middle of target sequence.
plot(recombination,xlab="normalized position",ylab="jump_counts")
vioplot(recombination$Freq)
```

Let's take a close look at the distribution of jump frequencies, and on the contrary, look at the conserved jump counts.

```{r, echo=FALSE, message=FALSE, warning=FALSE}
plot(recombination,ylim=c(0,400),xlab="normalized position",ylab="jump_counts")
lines(smooth.spline(recombination$recombination, recombination$Freq),col="blue")
hist(recombination$Freq,breaks=seq(0,400,1),axes=T,xlim=c(0,200),main ="Historgram of Freq ",xlab="frequency",ylab="position counts")
```



Then let's look at its barplot.

```{r, echo=FALSE, message=FALSE, warning=FALSE}
position <- as.numeric(levels(recombination$recombination)[recombination$recombination])
recombination$cut <- cut2(position,seq(0,1,0.1))
rec_cumsum <- ddply(recombination, "cut",
                   transform, cumsums=cumsum(Freq))
temp <- as.data.frame(table(recombination$cut))
temp$sumFreq <- temp$Freq
for (i in 2:10) { temp$sumFreq[i] <- temp$sumFreq[i] + temp$sumFreq[i-1] }
```


```{r}
barplot(rec_cumsum[temp$sumFreq,4],names.arg=temp[,1],density=20,border="red",col="blue",xlab="Normalized position",ylab="Number of jumps",space=0)
abline(v=c(4.9,6.3),col="yellow")
abline(v=c(2.5,4.8),col="darkgreen")
```

Now extract the HB5 location from Ghana seqs.
```{python}
##########extract HB5 positions in target sequence
import glob
import re
from mungo.fasta import FastaReader
from collections import defaultdict
import numpy as np
from __future__ import division
for h,s in FastaReader("/Users/fengqian/Downloads/UniMelb_shared-master/project/mosaic_data/Protein_translateable_pilot_upper_centroids.fasta"):
    L1=re.findall(r'(REDWW.*?\TC)',s);L2=re.findall(r'(REYWW.*?\TC)',s)
    if len("".join(L1))==0 and len("".join(L1))==0: continue
    elif len("".join(L1))!=0 and len("".join(L2))!=0:
        print ("extra HB5")
    else:
        L = L1+L2
        hb5_start = s.find("".join(L))+1
        hb5_end = hb5_start + len("".join(L))-1
        hb5_location = []
        hb5_location.append(hb5_start/len(s))
        hb5_location.append(hb5_end/len(s))
        with open("/Users/fengqian/Downloads/UniMelb_shared-master/project/mosaic_data/hb5_location_analysis.txt", 'a') as outfile:
            outfile.write("\t".join([str(c) for c in hb5_location])+"\n")
```

```{r, echo=FALSE, message=FALSE, warning=FALSE}
HB5 <- fread("/Users/fengqian/Downloads/UniMelb_shared-master/project/mosaic_data/hb5_location_analysis.txt"
                  , data.table = FALSE
                  , header=FALSE)
colnames(HB5)=c("left","right")
print("HB5 left endpoint location:")
summary(HB5$left)
print("HB5 right endpoint location:")
summary(HB5$right)
```


```{r}
HB3 <- unlist(fread("/Users/fengqian/Downloads/UniMelb_shared-master/project/mosaic_data/hb3_end_location.txt", data.table = FALSE)[1,])
```






```{r, echo=FALSE, message=FALSE, warning=FALSE}
ggplot(data=rec_cumsum, aes(x=recombination, y=cumsums)) +
  geom_bar(stat="identity")+
  geom_text(aes(y=cumsums, label=Freq), vjust=1.6, 
            color="yellow", size=3.5)+
  scale_fill_brewer(palette="Paired")+
  theme_minimal()
```





*********************************************************************************
*********************************************************************************
*********************************************************************************
*********************************************************************************


### Above method is a little overcomplicated. Let's replace by a simplier code. Global and Ghana figure will be integrated in one figure.

```{r, message=FALSE, warning=FALSE}
recombination_ghana <- unlist(fread("/Users/fengqian/Downloads/UniMelb_shared-master/project/mosaic_data/target_jumps_proportions.txt", data.table = FALSE)[1,])
recombination_global <- unlist(fread("/Users/fengqian/global_var/qian_report/mosaic_data/target_jumps_proportions.txt", data.table = FALSE)[1,])
recombination_position <- rbind(as.matrix(recombination_ghana),as.matrix(recombination_global))
df <- data.frame(f1=c(rep("Ghana",length(recombination_ghana)),rep("Global",length(recombination_global))),recombination_position=recombination_position,stringsAsFactors = FALSE)


pdf(file="/Users/fengqian/Downloads/UniMelb_shared-master/project/bkp_dist.pdf",width=10)#cols <- c("#F8766D","#00BFC4")
gg <- ggplot(aes(recombination_position), data = df)+geom_histogram(position = 'identity',alpha=0.5,aes(y = ..density..,fill = factor(f1)),color="black") +stat_density(geom = 'line',position = 'identity',aes(colour = factor(f1)),size=1)+ ylab("Density")+xlab("Normalized position")+scale_fill_manual(name="",values =c("#F8766D","#00BFC4"))+scale_color_manual(name="",values =  c("#F8766D","#00BFC4"))+ylim(0, 2.6)+xlim(0, 1)+ theme_bw()+theme(axis.text.x = element_text(colour ="black",size=18),axis.text.y = element_text(colour ="black",size=18),axis.title.y = element_text(size = 23,margin = unit(c(0, 4, 0, 0), "mm")),axis.title.x = element_text(size = 23,margin = unit(c(4, 0, 0, 0), "mm")))+ theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_rect(colour = "black",size=1))+theme(legend.title=element_blank(),legend.text=element_text(size=14))
gg
#
dev.off()
```


Alone show the Ghana jump position

```{r}
#pdf(file="Ghana_jump_proportion.pdf", height=4, width=6)
p <- qplot(recombination_ghana,geom="histogram",binwidth = 0.02,  xlab = "Normalized position", ylab = "Counts",
      col=I("black"), 
      fill=I("black"), 
      alpha=I(.2),
      xlim=c(0,1),
      ylim=c(0,1900))
p <- p +  geom_vline(aes(xintercept = mean(HB5$left)),linetype = 5,col='black',size=1)
p <- p +  geom_vline(aes(xintercept = mean(HB5$right)),linetype = 5,col='black',size=1)
p <- p +  geom_vline(aes(xintercept = 8/125),linetype = 5,col='black',size=1)
p <- p +  geom_vline(aes(xintercept = (125-7)/125),linetype = 5,col='black',size=1)
p <- p + annotate("text", x=0.57, y=1700, label= "HB5",size = 6) 
p <- p + annotate("text", x=0.01, y=1700, label= "HB3",size = 6)
p <- p + annotate("text", x=1.00, y=1700, label= "HB2",size = 6)
p <- p + annotate("text", x=1.00, y=1700, label= "HB2",size = 6)
p <- p + annotate("point", x = mean(HB5$left)-2*sd(HB5$left), y = 0,colour = "red", size = 1.5, alpha=0.4)
p <- p + annotate("point", x = mean(HB5$left)+2*sd(HB5$left), y = 0,colour = "red", size = 1.5, alpha=0.4)
p <- p + annotate("point", x = mean(HB5$right)-2*sd(HB5$right), y = 0,colour = "blue", size = 1.5, alpha=0.4)
p <- p + annotate("point", x = mean(HB5$right)+2*sd(HB5$right), y = 0,colour = "blue", size = 1.5, alpha=0.4)
p
#dev.off()
```


```{r}
# This chunk is for the manuscript. It show the bkp distribution for identified recombinant sequences.
#pdf(file="/Users/fengqian/Downloads/UniMelb_shared-master/project/recombination_detection/bkp_dist.pdf", height=4, width=6)
recombination_ghana_rec_only <- unlist(fread("/Users/fengqian/Downloads/UniMelb_shared-master/project/recombination_detection/target_jumps_proportions_for_recombinants.txt", data.table = FALSE)[1,])
p <- qplot(recombination_ghana_rec_only,geom="histogram",binwidth = 0.02, xlab = "Normalized position", ylab = "Frequency",
      col=I("black"), 
      fill=I("black"), 
      alpha=I(.2),
      xlim=c(0,1),
      ylim=c(0,1900))
p <- p +  geom_vline(aes(xintercept = mean(HB5$left)),linetype = 5,col='black',size=1)
p <- p +  geom_vline(aes(xintercept = mean(HB5$right)),linetype = 5,col='black',size=1)
p <- p +  geom_vline(aes(xintercept = 8/125),linetype = 5,col='black',size=1)
p <- p +  geom_vline(aes(xintercept = (125-7)/125),linetype = 5,col='black',size=1)
p <- p + annotate("text", x=0.57, y=1700, label= "HB5",size = 6) 
p <- p + annotate("text", x=0.01, y=1700, label= "HB3",size = 6)
p <- p + annotate("text", x=1.00, y=1700, label= "HB2",size = 6)
p <- p + annotate("text", x=1.00, y=1700, label= "HB2",size = 6)
p <- p + annotate("point", x = mean(HB5$left)-2*sd(HB5$left), y = 0,colour = "red", size = 1, alpha=0.4)
p <- p + annotate("point", x = mean(HB5$left)+2*sd(HB5$left), y = 0,colour = "red", size = 1, alpha=0.4)
p <- p + annotate("point", x = mean(HB5$right)-2*sd(HB5$right), y = 0,colour = "blue", size = 1, alpha=0.4)
p <- p + annotate("point", x = mean(HB5$right)+2*sd(HB5$right), y = 0,colour = "blue", size = 1, alpha=0.4)
p
#dev.off()
```




Show Ghana jump position distribution:
```{r}
hist(recombination_ghana,breaks=seq(0,1,by=0.02))
#plot(density(recombination))
#barplot(table(cut(recombination_ghana,seq(0,1,by=0.02))))
```

```{r}
recombination_position_freq <- rbind(as.matrix(as.data.frame(table(recombination_ghana))$Freq),as.matrix(as.data.frame(table(recombination_global))$Freq))
df <- data.frame(f1=c(rep("Ghana",length(as.matrix(as.data.frame(table(recombination_ghana))$Freq))),rep("global",length(as.matrix(as.data.frame(table(recombination_global))$Freq)))),recombination_position_freq=recombination_position_freq,stringsAsFactors = FALSE)
pdf(file="Ghana_jump_position_proportion.pdf", height=4, width=6)
p <- ggplot(df, aes(factor(f1), recombination_position_freq))
p <- p + geom_violin(aes(fill = factor(f1)),draw_quantiles = c(0.25, 0.5, 0.75))
p <- p + xlab(" ")
p <- p + ylab("Counts of each position")
p <- p + scale_fill_discrete(name = "Location")
p
dev.off()
```



### Let's compare jump positions for all sequences and jump once sequences in Ghana pilot dataset

```{r, message=FALSE, warning=FALSE}
recombination_ghana <- unlist(fread("/Users/fengqian/Downloads/UniMelb_shared-master/project/mosaic_data/target_jumps_proportions.txt", data.table = FALSE)[1,])
recombination_ghana_jumponce <- unlist(fread("/Users/fengqian/Downloads/UniMelb_shared-master/project/mosaic_data/target_jumps_proportions(onejumpseqs).txt", data.table = FALSE)[1,])
recombination_position <- rbind(as.matrix(recombination_ghana),as.matrix(recombination_ghana_jumponce))
df <- data.frame(f1=c(rep("Ghana_all_seqs",length(recombination_ghana)),rep("Ghana_jumponce_seqs",length(recombination_ghana_jumponce))),recombination_position=recombination_position,stringsAsFactors = FALSE)


#cols <- c("#F8766D","#00BFC4")
gg <- ggplot(aes(recombination_position), data = df)
gg <- gg + geom_histogram(position = 'identity',
    alpha=0.3,
    aes(y = ..density..,
    fill = factor(f1))) +
    stat_density(geom = 'line',
    position = 'identity',
    aes(colour = factor(f1)))
gg <- gg + ylab("Density")
gg <- gg + xlab("Normalized position")
gg <- gg + scale_fill_manual(name="",values =  c("#F8766D","#00BFC4"))
gg <- gg + scale_color_manual(name="",values =  c("#F8766D","#00BFC4"))
gg <- gg +  ylim(0, 3.7)
gg <- gg +  xlim(0, 1)
gg

```






