---
title: "Sup_ghana_1"
author: "Qian Feng"
date: "2018/7/1"
output:
  pdf_document: default
---

```{r global_options, include=FALSE}
knitr::opts_chunk$set(fig.width=12, fig.height=8,
                      echo=TRUE, warning=FALSE, message=FALSE,
                      tidy=TRUE)
```


##Load libraries

```{r, message=FALSE, warning=FALSE}
library(data.table)
library(dplyr)
library(ggfortify)
library(ggplot2)
#library(Rtsne)
library(stringr)
library(starmie)
library(ggtree)
library(ape)
library(pheatmap)
library(proxy)
library(stringr)
library(knitr)
library(plotly)
library(mixOmics)
cols <- c("#a6cee3","#1f78b4","#b2df8a","#33a02c","#fb9a99","#e31a1c","#fdbf6f","#ff7f00","#cab2d6","#6a3d9a","#ffff99","#b15928","#8dd3c7","#ffffb3","#bebada","#fb8072","#80b1d3","#fdb462","#b3de69","#fccde5","#d9d9d9","#bc80bd","#ccebc5","#ffed6f")
```

Note that my project directory is in "/Users/fengqian/Downloads/phd/UniMelb_shared-master/project". When you replicate my analysis, you need to change to your own dir.

We start by clustering the raw Pilot reads using a python script that makes use of the Usearch software suite.
```{python, eval=FALSE, include=FALSE}
from mungo.fasta import FastaReader
from collections import defaultdict
with open("/Users/fengqian/Downloads/phd/UniMelb_shared-master/project/OTU/pilot_upper.fasta",'w') as outfile:
    for h,seq in FastaReader("/Users/fengqian/Downloads/phd/UniMelb_shared-master/project/sequences/Ghana/Pilot.fasta"):
        seq=seq.upper()
        outfile.write(">"+h+"\n"+seq+"\n")
```

```{bash, eval = FALSE}
cd
Python /Users/fengqian/Downloads/phd/UniMelb_shared-master/project/scripts/clusterDBLa.py -o /Users/fengqian/Downloads/phd/UniMelb_shared-master/project/OTU/ -r /Users/fengqian/Downloads/phd/UniMelb_shared-master/project/OTU/pilot_upper.fasta
```

####################################################################################################
### Binary Analysis
Now we can investigate the isolates based on shared DBLa sequence types. We have extracted 161 isolates from file "Pilot.fasta"(35591 reads), and prveious isolate_information.csv provides the locations of some of isolates (137 isolates,less than 161. At last, 133 out of 161 isolates have location), let's add these location information.
```{r}
isolateInformation <- fread("/Users/fengqian/Downloads/phd/UniMelb_shared-master/project/data/ghana_isolateInformation_nolab.csv"
                            , header=TRUE
                            , data.table = FALSE)
#Add in location information
isolatepilot <- fread("/Users/fengqian/Downloads/phd/UniMelb_shared-master/project/data/isolateInformation.csv"
                            , header=TRUE
                            , data.table = FALSE)
isolatepilot <- isolatepilot[isolatepilot$Publication=="DayLab_Ghana_Pilot",]
isolatepilot$Isolate <- unlist(lapply(isolatepilot$Isolate
                                            , function(x) {
                                              paste("P",str_split(x,  "_")[[1]][[1]],sep="")}))
isolateInformation <- merge(isolateInformation,isolatepilot,by="Isolate",all.x=TRUE)

isolateInformation <- isolateInformation[isolateInformation$Survey=="pilot",]
isolateInformation$Location[which(isolateInformation$Location %in% NA)]="Ghana_Unknown"
isolateInformation$Location=unlist(lapply(isolateInformation$Location
                                            , function(x) {
                                              str_split(x,  "_")[[1]][[2]]}))#concise the location for thesis
isolateInformation$Location[isolateInformation$Location=="VeaGowrie"] <- "Vea/Gowrie"

otuTable <- fread("/Users/fengqian/Downloads/phd/UniMelb_shared-master/project/OTU/pilot_upper_renamed_otuTable_binary.txt"
                  , data.table = FALSE
                  , header=TRUE)

otuMatrix <- as.matrix(otuTable[,2:ncol(otuTable)])
rownames(otuMatrix) <- otuTable$`#OTU ID`
```

We found a total of `r sum(otuMatrix)` (35566) reads in the combined dataset, which clustered into a total of `r nrow(otuMatrix)` (17923). Of these `r sum(rowSums(otuMatrix)==1)`(11607) were only seen in one isolate.


We next perform some filtering. We only investigate isolates that were found to have more than 20 DBLa types. This was found to be a sensible threshold on having adequetly sequences an isolates VAR repetoir. Furthermore as we are interested in the realtionship between isolates, we exclude the singletons from the binary analysis.
```{r}
#Filter otus that only appear in one isolate and isolates with less than 20 types
MIN_ISOLATE_PER_OTU = 2
MIN_OTUS_PER_ISOLATE = 20
MAX_OTUS_PER_ISOLATE = Inf
otuMatrix <- otuMatrix[, colSums(otuMatrix) >= MIN_OTUS_PER_ISOLATE]
otuMatrix <- otuMatrix[, colSums(otuMatrix) <= MAX_OTUS_PER_ISOLATE]
otumatrixfiltered <- otuMatrix[rowSums(otuMatrix) >= MIN_ISOLATE_PER_OTU, ]
colnames(otumatrixfiltered) <-  unlist(lapply(colnames(otumatrixfiltered)
                                            , function(x) {
                                              str_split(x,  ".MID")[[1]][[1]]}))
```


We can now look at the number of reads per isolate for the different locations in ghana.
```{r}
otu_sums <- data.frame(Isolate=colnames(otumatrixfiltered), num_otus=colSums(otumatrixfiltered)
                       , stringsAsFactors = FALSE)
  
otu_sums <- merge(otu_sums, isolateInformation, by.x='Isolate', by.y='Isolate'
                , all.x=TRUE)

#median_summary <- otu_sums %>% group_by(Location) %>% 
  #summarise(n=n(),
            #median=median(num_otus),
            #max=max(num_otus))
  
#boxplot plot
pdf(file="Types_per_isolate.pdf",width=10)
gg <- ggplot(otu_sums, aes(factor(Location), num_otus, fill=Location)) + geom_boxplot()+scale_color_manual(values = cols[1:length(unique(isolateInformation$Location))])+xlab("")+ylab("Number of types per isolate")+ theme_bw()+theme(axis.text.x = element_text(colour ="black",size=18),axis.text.y = element_text(colour ="black",size=18),axis.title.y = element_text(size = 23,margin = unit(c(0, 4, 0, 0), "mm")),axis.title.x = element_text(size = 23,margin = unit(c(4, 0, 0, 0), "mm")))+ theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_rect(colour = "black",size=1))+ theme(legend.position = "none")
#+theme(legend.title = element_blank(),legend.position = "top",legend.text=element_text(size=14),legend.background = element_blank(),legend.box.background = element_rect(colour = "white"))
gg
dev.off()


pdf(file="Types_per_isolate.pdf",width=10)#this figure is for examiners.It's the same with above one.
ggplot(otu_sums, aes(factor(Location), num_otus)) + geom_boxplot(col=I("black"),fill=I("black"),alpha=I(.4))+xlab("")+ylab("Number of types per isolate")+scale_y_continuous(breaks=seq(0,600,100),limits=c(0,618))+theme_bw()+theme(axis.text.x = element_text(colour ="black",size=18),axis.text.y = element_text(colour ="black",size=18),axis.title.y = element_text(size = 23,margin = unit(c(0, 4, 0, 0), "mm")),axis.title.x = element_text(size = 23,margin = unit(c(4, 0, 0, 0), "mm")))+ theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_rect(colour = "black",size=1))+ theme(legend.position = "none")+stat_summary(fun=mean, geom="point", shape=20, size=12, color="red", fill="red")
#+ geom_segment(aes(x = 1, y = 350, xend = 2, yend = 350))+ geom_segment(aes(x = 2, y = 450, xend = 3, yend = 450))+ geom_segment(aes(x = 1, y = 550, xend = 3, yend = 550))+ geom_segment(aes(x = 1, y = 350, xend = 1, yend = 340))+ geom_segment(aes(x = 2, y = 350, xend = 2, yend = 340))+ geom_segment(aes(x = 2, y = 450, xend = 2, yend = 440))+ geom_segment(aes(x = 3, y = 450, xend = 3, yend = 440))+ geom_segment(aes(x = 3, y = 550, xend = 3, yend = 540))+ geom_segment(aes(x = 1, y = 550, xend = 1, yend = 540))+scale_size_identity()+ annotate(geom="text", x=1.5, y=375, label="p = 0.428",size=6)+annotate(geom="text", x=2.5, y=475, label="p = 0.660",size=6)+annotate(geom="text", x=2, y=575, label="p = 0.109",size=6)
dev.off()
```



```{r}
x=otu_sums$num_otus[which(otu_sums$Location=="Unknown")]
y=otu_sums$num_otus[which(otu_sums$Location=="Vea/Gowrie")]
z=otu_sums$num_otus[which(otu_sums$Location=="Soe")]
wilcox.test(z, y, alternative = "two.sided")
wilcox.test(z, x, alternative = "two.sided")
wilcox.test(y, x, alternative = "two.sided")
```




## PCA
```{r}
otuMatrixfiltered_t <- t(otumatrixfiltered)

pca <- prcomp(otuMatrixfiltered_t)

pca <- data.frame(Isolate = rownames(otuMatrixfiltered_t)
                     , pca$x[, 1:6]
                     , stringsAsFactors = FALSE)

pca <- merge(pca, isolateInformation, by.x='Isolate', by.y='Isolate'
                , all.x=TRUE)

#PCA plot
pdf(file="PCA1.pdf")
gg <- ggplot(pca, aes(PC1, PC2, colour=Location)) + geom_point(size=3)+xlab("PC1")+ylab("PC2")+ylim(-4, 4)+ theme_bw()+theme(axis.text.x = element_text(colour ="black",size=18),axis.text.y = element_text(colour ="black",size=18),axis.title.y = element_text(size = 23,margin = unit(c(0, 4, 0, 0), "mm")),axis.title.x = element_text(size = 23,margin = unit(c(4, 0, 0, 0), "mm")))+ theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_rect(colour = "black",size=1))+ theme(legend.position = "top",legend.title = element_blank(),legend.text=element_text(size=16),legend.background = element_blank(),legend.box.background = element_rect(colour = "white")) 
#gg <- gg + scale_color_manual(values = cols[1:length(unique(isolateInformation$Location))])
gg
dev.off()
pdf(file="PCA2.pdf")
gg <- ggplot(pca, aes(PC1, PC3, colour=Location)) + geom_point(size=3)+xlab("PC1")+ylab("PC3")+ylim(-4, 4)+ theme_bw()+theme(axis.text.x = element_text(colour ="black",size=18),axis.text.y = element_text(colour ="black",size=18),axis.title.y = element_text(size = 23,margin = unit(c(0, 4, 0, 0), "mm")),axis.title.x = element_text(size = 23,margin = unit(c(4, 0, 0, 0), "mm")))+ theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_rect(colour = "black",size=1))+ theme(legend.position = "top",legend.title = element_blank(),legend.text=element_text(size=16),legend.background = element_blank(),legend.box.background = element_rect(colour = "white")) 
#gg <- gg + scale_color_manual(values = cols[1:length(unique(isolateInformation$Location))])
gg
dev.off()

pdf(file="PCA.pdf",width=10)
gg <- ggplot(pca, aes(PC1, PC2, colour=Location)) + geom_point(size=3)+xlab("PC1")+ylab("PC2")+ylim(-4, 4)+scale_x_continuous(breaks=seq(-4,8,4),limits=c(-4,8))+theme_bw()+theme(axis.text.x = element_text(colour ="black",size=18),axis.text.y = element_text(colour ="black",size=18),axis.title.y = element_text(size = 23,margin = unit(c(0, 4, 0, 0), "mm")),axis.title.x = element_text(size = 23,margin = unit(c(4, 0, 0, 0), "mm")))+ theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_rect(colour = "black",size=1))+ theme(legend.position = "top",legend.title = element_blank(),legend.text=element_text(size=18),legend.background = element_blank(),legend.box.background = element_rect(colour = "white")) 
#gg <- gg + scale_color_manual(values = cols[1:length(unique(isolateInformation$Location))])
gg
dev.off()

```



## Conserved DBLa types

We can also investigate the most conserved DBLa types. First let's look at a histogram of the number of times each DBLa is seen in the global population. It suggests that the majority of types are seen less than 20 times(`r length(which(rowSums(otumatrixfiltered)<20))`>>`r length(which(rowSums(otumatrixfiltered)>=20))`). We then take a closer look at those seen at least 20 times.

```{r}
pdf(file="Ghana_typeoccurrence.pdf",width=11)
temp=as.data.frame(table(rowSums(otumatrixfiltered)))
ggplot(data=temp, aes(x=as.numeric(as.character(Var1)), y=Freq)) +
    geom_bar(stat="identity",col=I("black"), 
      fill=I("black"), 
      alpha=I(.4))+xlab(expression(paste("Frequency of each DBL",alpha, " type") ))+ylab("Counts")+scale_y_continuous(breaks=seq(0,3400,200),limits=c(0,3400))+scale_x_continuous(breaks=seq(0,130,by=10),labels=as.character(seq(0,130,by=10)))+theme_bw()+theme(axis.text.x = element_text(colour ="black",size=18),axis.text.y = element_text(colour ="black",size=18),axis.title.y = element_text(size = 23,margin = unit(c(0, 4, 0, 0), "mm")),axis.title.x = element_text(size = 23,margin = unit(c(4, 0, 0, 0), "mm")))+ theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_rect(colour = "black",size=1))
dev.off()
#expand = c(0, 0),expand = c(0, 0),
```





```{r, message=FALSE, warning=FALSE}
h <- hist(rowSums(otumatrixfiltered), breaks=500, plot = FALSE)
plot(h$mids, h$counts, log="y")##h$mids means the rowSums(otumatrixfiltered)
abline(v=c(20,50), col='red')
```

```{r}
majorTypeMatrix <- otumatrixfiltered[rowSums(otumatrixfiltered)>=20,]
col_annotations <- data.frame(Isolate = colnames(majorTypeMatrix),
                              stringsAsFactors = FALSE)
col_annotations <- merge(col_annotations, isolateInformation,
                         by.x="Isolate", by.y="Isolate",
                         all.x=TRUE)
rownames(col_annotations) <- col_annotations$Isolate
col_annotations <- col_annotations[, c("Isolate","Location")]
col_annotations <- col_annotations[order(col_annotations$Location),]
majorTypeMatrix <- majorTypeMatrix[, match(col_annotations$Isolate, colnames(majorTypeMatrix))]
col_annotations$Isolate <- NULL
#pdf(file="heatmap.pdf",width=11)
pheatmap(majorTypeMatrix, cluster_cols = FALSE
         , annotation_col = col_annotations
         , show_rownames = FALSE
         , fontsize_row=2
         , show_colnames = FALSE
         , treeheight_row = 0
         , color=c("#ffffff","#000000"),legend = FALSE,annotation_names_col=FALSE,annotation_colors=list(Location=c("Soe"="#F8766D","Unknown"="#00BA38","Vea/Gowrie"="#619CFF")))
#dev.off()

```

This suggests three regions seem to have no big difference about the distribution of DBLa tags again.


## RAxML
First we need to produce a binary fasta file:

```{r echo=FALSE, message=FALSE, warning=FALSE}
text=ggplot2:::interleave(paste(">", rownames(otuMatrixfiltered_t), sep=""),
                          apply(otuMatrixfiltered_t, 1 , paste , collapse = "" ))
writeLines(text, con="/Users/fengqian/Downloads/phd/UniMelb_shared-master/project/OTU/DBLa_binary.fasta",
           sep="\n")
```

Now we can run RAxML. The commands are from RAxML tutorial (https://cme.h-its.org/exelixis/web/software/raxml/hands_on.html).
```{bash, eval=FALSE, include=FALSE}
cd /data/gpfs/projects/punim0609/qian_feng/UniMelb_shared-master/project/standard-RAxML-master

./raxmlHPC-PTHREADS -m BINCAT -p 12345 -# 20 -s /data/gpfs/projects/punim0609/qian_feng/UniMelb_shared-master/project/OTU/DBLa_binary.fasta -n T1 -T 10

./raxmlHPC-PTHREADS -m BINCAT -p 12345 -b 12345 -# 100 -s /data/gpfs/projects/punim0609/qian_feng/UniMelb_shared-master/project/OTU/DBLa_binary.fasta -n T2 -T 10

./raxmlHPC-PTHREADS -m BINCAT -p 12345 -f b -t RAxML_bestTree.T1 -z RAxML_bootstrap.T2 -n T3


cp RAxML_bipartitionsBranchLabels.T3 ../OTU
```


We can now look at the output of RAxML.
```{r}
raxml <- read.raxml("/Users/fengqian/Downloads/phd/UniMelb_shared-master/project/OTU/RAxML_bipartitionsBranchLabels.T3")
groupInfo <- isolateInformation %>% group_by(Location) %>%
  do(taxa_list = .$Isolate)

groups <- lapply(groupInfo$taxa_list, as.vector)
names(groups) <- groupInfo$Location
raxml <- groupOTU(raxml, groups)

pdf(file="/Users/fengqian/Downloads/phd/UniMelb_shared-master/project/OTU/tree.pdf",width=10)
#gg <- ggtree(raxml, aes(color=group, label=node),size=0.3, branch.length = "none", layout="circular") + geom_text2(aes(x = branch,label=bootstrap,subset = !is.na(as.numeric(bootstrap)) & as.numeric(bootstrap) > 60))
gg <- ggtree(raxml, aes(color=group, label=node),size=0.3) + geom_text2(aes(x = branch,label=bootstrap,subset = !is.na(as.numeric(bootstrap)) & as.numeric(bootstrap) > 60))
gg <- gg +  theme(legend.position="right")
gg <- gg +  labs(color='Location')
gg <- gg + theme(legend.text=element_text(size=14),legend.title = element_text(size = 18))
gg
dev.off()
```




## FFP

Try pilot data still.
We employ the centroids P3D7.fasta to get the appropriate k.
```{bash, eval=FALSE, include=FALSE}
cd
Python /Users/fengqian/Downloads/phd/UniMelb_shared-master/project/scripts/clusterDBLa.py -o /Users/fengqian/Downloads/phd/UniMelb_shared-master/project/ffp_data/ -r /Users/fengqian/Downloads/phd/UniMelb_shared-master/project/data/P3D7.fasta
```


```{bash, eval=FALSE, include=FALSE}
cd ffp-3.19
ffpvprof -e 40 -f 2 /Users/fengqian/Downloads/phd/UniMelb_shared-master/project/ffp_data/P3D7_renamed_centroids.fasta >/Users/fengqian/Downloads/phd/UniMelb_shared-master/project/ffp_data/ffp_word_usage.txt

ffpreprof -e 40  /Users/fengqian/Downloads/phd/UniMelb_shared-master/project/ffp_data/P3D7_renamed_centroids.fasta >/Users/fengqian/Downloads/phd/UniMelb_shared-master/project/ffp_data/ffp_entropy_profile.txt
```


First let look at word usage to get an idea of a lower bound.The peak is in feature count distribution is our lower bound, termed as $k_{hman}$,Then from the cumulative relative entropy curve, get the minimal $k_{CREmin}$ is our upper bound. 

We can now attempt to choose an appropriate value of k. First let look at word usage to get an idea of a lower bound.
```{r}
word_usage <- fread("/Users/fengqian/Downloads/phd/UniMelb_shared-master/project/ffp_data/ffp_word_usage.txt",
                    data.table = FALSE)
plot(word_usage,xlab="Feature length, k",ylab="Feature count",type="h",col="blue")
```


```{r}
entropy <- fread("/Users/fengqian/Downloads/phd/UniMelb_shared-master/project/ffp_data/ffp_entropy_profile.txt",
                    data.table = FALSE)
plot(entropy,xlab="Feature length, k",ylab="Cumulative relative entropy, CRE",pch=15,type="o",col="dark red")
```

Thus a choice of k=20 appears to be appropriate.

We can now run a script to calculate the ffp distance matrix, its output is .phylip format.
```{bash, eval=FALSE}
 python /vlsci/SG0011/qian-feng/UniMelb_shared-master/project/scripts/ffp.py --kmer_length 20 --out /vlsci/SG0011/qian-feng/UniMelb_shared-master/project/ffp_data/ffp_distance_matrix.phylip --seq /vlsci/SG0011/qian-feng/UniMelb_shared-master/project/sequences/Ghana/Pilot.fasta --verbos
```


Finally a tree was built using fastme v2.1.5 with default parameters. We can now have a look at the resulting tree.
```{r}
ffp <- read.tree("/Users/fengqian/Downloads/phd/UniMelb_shared-master/project/ffp_data/ffp_distance_matrix_phylip_fastme/ffp_distance_matrix_phylip_fastme-tree.txt")

gg <- ggtree(ffp, size=0.3, branch.length = "none", layout="circular") + ggtitle("Pilot Phylogenetic tree") 
gg

```


```{r}
###add location information
groupInfo <- isolateInformation %>% group_by(Location) %>%
  do(taxa_list = .$Isolate)

groups <- lapply(groupInfo$taxa_list, as.vector)
names(groups) <- groupInfo$Location
ffp <- groupOTU(ffp, groups)

t_cols <- cols
names(t_cols) <- names(groups)
pdf(file="/Users/fengqian/Downloads/phd/UniMelb_shared-master/project/ffp_data/ffptree.pdf",width=10)
#gg <- ggtree(ffp, aes(color=group, label=node), size=0.3, branch.length = "none", layout="circular")
#gg <- gg + scale_color_manual(values=as.character(t_cols[names(t_cols) %in% names(groups)]),labels=names(t_cols)) 
gg <- ggtree(ffp, aes(color=group, label=node), size=0.3)
gg <- gg +  theme(legend.position="right")
gg <- gg +  labs(color='Location')
gg <- gg + theme(legend.text=element_text(size=14),legend.title = element_text(size = 18))
gg
dev.off()
```


compare the Raxml and ffp trees:
```{r echo=FALSE, warning=FALSE}
treedist(ffp,raxml,check.labels=FALSE)
```
Actually, comparing the resulting trees from RAxML and FFP, if we ignore the isolate's location information, there is still a big difference about these two trees' topologies.This makes sense since different methods always produce different trees. Whatever method is employed, the separation about the location is always not clearly.


Examiner wants to see a statistical test to check whether the trees are not separated by catment areas statistically.
```{r}
source("/Users/fengqian/Downloads/phd/UniMelb_shared-master/project/delta_statistic-master/code.R")
#ffp tree
ffp <- read.tree("/Users/fengqian/Downloads/phd/UniMelb_shared-master/project/ffp_data/ffp_distance_matrix_phylip_fastme/ffp_distance_matrix_phylip_fastme-tree.txt")
plot(ffp)

unknown_tip= isolateInformation$Isolate[isolateInformation$Location=="Unknown"]
ffp_two_loc=drop.tip(ffp, unknown_tip)
ffp_1=multi2di(ffp_two_loc)
ffp_1$edge.length[ffp_1$edge.length==0] <- quantile(ffp_1$edge.length,0.1)*0.1
trait <-isolateInformation$Location[match(ffp_1$tip.label,isolateInformation$Isolate)]
deltaA <- delta(trait,ffp_1,0.1,0.0589,10000,10,100)#2.559353
random_delta <- rep(NA,100)
for (i in 1:100){
  rtrait <- sample(trait)
  random_delta[i] <- delta(rtrait,ffp_1,0.1,0.0589,10000,10,100)
}
#random_delta <- random_delta[!is.na(random_delta)]
p_value <- sum(random_delta>deltaA)/length(random_delta)
p_value #0.18 indicates there is no evidence for phylogenetic signal.
```


```{r}
#raxml tree
raxml <- read.raxml("/Users/fengqian/Downloads/phd/UniMelb_shared-master/project/OTU/RAxML_bipartitionsBranchLabels.T3")
raxml_tree=raxml@phylo
raxml_two_loc=drop.tip(raxml_tree, unknown_tip)
raxml_1=multi2di(raxml_two_loc)
trait <-isolateInformation$Location[match(raxml_1$tip.label,isolateInformation$Isolate)]
deltaA <- delta(trait,raxml_1,0.1,0.0589,10000,10,100)
random_delta <- rep(NA,100)
for (i in 1:100){
  rtrait <- sample(trait)
  random_delta[i] <- delta(rtrait,raxml_1,0.1,0.0589,10000,10,100)
}
random_delta <- random_delta[!is.na(random_delta)]
p_value <- sum(random_delta>deltaA)/length(random_delta)
p_value #0.84 also indicates there is no evidence for phylogenetic signal.
```

####################################################################################################

